cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -697,16 +703,15 @@ private V1PodSpec createPodSpec(Location runtimeConfigLocation,
     volumeMounts.add(new V1VolumeMount().name("workdir").mountPath(workDir));
     volumeMounts.addAll(Arrays.asList(extraMounts));
 
-    // Setup the container environment. Inherit everything from the current pod.
-    Map<String, String> environs = podInfo.getContainerEnvironments().stream()
-      .collect(Collectors.toMap(V1EnvVar::getName, V1EnvVar::getValue));
-    // Add all environments for the runnable
-    environs.putAll(environments.get(runnableName));
```

Comment:
It seems like this block of code was just moved down a couple lines and this line (704) was deleted. Is that intentional?

Reply:
Thanks for catching this. _environs.putAll(environments.get(runnableName))_ was moved _createContainers_ method. I though noticed that we also need it for the init container.